src/javascript/app/pages/user/new_account/welcome_page.js

```diff
@@ -1,45 +1,53 @@
 const BinarySocket     = require('../../../base/socket');
 const Client           = require('../../../base/client');
 const getElementById   = require('../../../../_common/common_functions').getElementById;
-const localize         = require('../../../../_common/localize').localize;
 const Url              = require('../../../../_common/url');
-const createElement    = require('../../../../_common/utility').createElement;
 const showLoadingImage = require('../../../../_common/utility').showLoadingImage;
 
 const WelcomePage = (() => {
+
+    let el_welcome_container,
+        is_virtual,
+        upgrade_info;
+
+    const init = () => {
+        upgrade_info      = Client.getUpgradeInfo();
+        is_virtual        = Client.get('is_virtual');
+        el_welcome_container  = getElementById('welcome_container');
+    };
+
+    const getCanUpgrade = (upgrade_type ,  { can_upgrade_to } = upgrade_info) => can_upgrade_to.includes(upgrade_type);
```

Comment:
(throughout) can we use `ClientBase.getBasicUpgradeInfo` method instead of this?

Reply:
done